### PR TITLE
v6 beta branch

### PR DESCRIPTION
**Pull request by yoshuawuyts**
in-progress target branch we can use for breaking behavior. The prototype branch is basically a full rewrite; this allows us to put out some betas and iterate on the minor changes we want for the next version.

Basically I'm thinking we keep the API stable, but yeah there's a few nits that are technically breaking. So we should iron those out in a semver major. Yay for boring changes :sparkles:

### related issues
- [x] call DOMcontentloaded on next frame https://github.com/yoshuawuyts/choo/pull/465
- [x] use nanotiming https://github.com/yoshuawuyts/choo/pull/476
- [x] become prototype based https://github.com/yoshuawuyts/choo/pull/480
- [x] add tracing to nanobus https://github.com/yoshuawuyts/nanobus/pull/10
- [x] revert special casing of root onload handlers https://github.com/yoshuawuyts/nanomorph/pull/65
- [x] fix select tags https://github.com/yoshuawuyts/nanomorph/pull/60
- [x] remove onload https://github.com/shama/bel/pull/69
- [x] should clear performance timers https://github.com/yoshuawuyts/choo/issues/461
- [x] remove nanomount https://github.com/yoshuawuyts/choo/issues/481
- [x]  remove nanohistory (too small a module)
- [x] fix whitespace in `bel`
- [ ] fix whitespace in `yo-yoify` https://github.com/shama/yo-yoify/issues/49
- [ ] fix whitespace in `pelo` https://github.com/shuhei/pelo/issues/7
- [ ] upgrading bankai to new `yo-yoify`
- [ ] upgrading `bel` to new `pelo`

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
Timing API has been upgraded, now shows separate timings for creating DOM trees + morphing the trees.

screen shot 2017-05-09 at 13 35 01

I'm not sure about the event names tho; right now they are:
```txt
choo.render:view.route   # for choo's internal stuff
choo.emitter:foo:bar     # for an event a user registers
```
What I'm wondering is if we should perhaps upgrade [nanobus](https://github.com/yoshuawuyts/nanobus/) to delimit internal events with a `"/"` so it'd become:

```txt
choo:render/view:route   # for choo's internal stuff
choo:emitter/foo:bar     # for an event a user registers
```

Thoughts? cc/ @juliangruber @timwis

**Comment by yoshuawuyts** [comment]
note to self: move the new choo `performance.measure()` to `requestIdleCallback` and use https://github.com/yoshuawuyts/nanotiming/pull/1

**Comment by juliangruber** [comment]
thoughts:

- `[choo] render: view.route`
- `choo.render(view.route)`
- `[choo] render(view.route)`

**Comment by moszeed** [comment]
i like the following proposals by @juliangruber 
- choo.render(view.route)
- [choo] render(view.route)

**Comment by yoshuawuyts** [comment]
@juliangruber gonna try some of those out; like the `choo.render(view.route)` idea for sure! - should become part of https://github.com/yoshuawuyts/nanotiming/pull/3 I reckon

**Comment by yoshuawuyts** [comment]
Also thinking of adding a `navigate` event which is fired whenever path changes, and returns the current name of the route. This could be used in the logger as a `debug` style event - and would be useful for APM providers so they can hook into the actual name of the route.

**Comment by tornqvist** [comment]
Is there any reason for `createLocation` to not be exposed on the prototype? It would kind of solve #440 as you could roll your own method that ignores hashes and then simply set `opts.href` to false and implement nanohref in a similar way.

**Comment by yoshuawuyts** [comment]
@tornqvist I think what you're proposing makes sense (:

**Comment by yoshuawuyts** [comment]
Last issues remaining:
- https://github.com/shama/yo-yoify/issues/49
- https://github.com/shuhei/pelo/issues/7
- upgrading bankai to new `yo-yoify`
- upgrading `bel` to new `pelo`

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+This release also includes an overhauled timing API. We're now capturing more
+events, and are leveraging this in tools like `choo-log` to provide consistent
+feedback on performance.
+
+Choo is now also completely prototype based. It not only means faster execution
+times, and more maintainable codebase - but you can also override methods if
+you ever need to.
+
+We've also tweaked the event system a little. All built-in event names are now
+available as constants under `choo.events`. And we've introduced three new
```
Comment (juliangruber) [inline CHANGELOG.md]: three or two?

**Comment by yoshuawuyts** [comment]
Oops, good point - it's two brand new ones, and one that was introduced a
couple of weeks ago and folks might not be aware of. Will fix! :D

File: CHANGELOG.md (hunk 1 above)
Comment (yoshuawuyts) [inline CHANGELOG.md]: Oops, good point - it's two brand new ones, and one that was introduced a
couple of weeks ago and folks might not be aware of. Will fix! :D

Comment (yoshuawuyts) [inline CHANGELOG.md]: Fixed!

**Comment by yoshuawuyts** [comment]
📦 v6.0.0-0 (release candidate)

Please share any feedback here https://github.com/yoshuawuyts/choo/issues/519. Thanks! :tada:

**Pull request merged**